extensions/standard-processors/tests/unit/YamlConnectionParserTest.cpp

```diff
@@ -0,0 +1,200 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "core/yaml/YamlConnectionParser.h"
+
+#include "core/yaml/YamlConfiguration.h"
+#include "TailFile.h"
+#include "TestBase.h"
+#include "utils/TestUtils.h"
+
+namespace {
+
+using org::apache::nifi::minifi::core::yaml::YamlConnectionParser;
+using org::apache::nifi::minifi::core::YamlConfiguration;
+using RetryFlowFile = org::apache::nifi::minifi::processors::TailFile;
+
+TEST_CASE("Connections components are parsed from yaml.", "[YamlConfiguration]") {
+  const std::shared_ptr<logging::Logger> logger = logging::LoggerFactory<YamlConfiguration>::getLogger();
+  core::ProcessGroup parent(core::ProcessGroupType::ROOT_PROCESS_GROUP, "root");
+  gsl::not_null<core::ProcessGroup*> parent_ptr{ &parent };
+
+  SECTION("Source relationships are read") {
+    const auto connection = std::make_shared<minifi::Connection>(nullptr, nullptr, "name");
+    std::string serialized_yaml;
+    std::vector<std::string> expectations;
+    SECTION("Single relationship name") {
+      serialized_yaml = std::string { "source relationship name: success\n" };
+      expectations = { "success" };
+    }
+    SECTION("List of relationship names") {
+      serialized_yaml = std::string {
+          "source relationship names:\n"
+          "- success\n"
+          "- failure\n"
+          "- something_else\n" };
+      expectations = { "success", "failure", "something_else" };
+    }
+    YAML::Node connection_node = YAML::Load(serialized_yaml);
+    YamlConnectionParser yaml_connection_parser(connection_node, "test_node", parent_ptr, logger);
+    yaml_connection_parser.configureConnectionSourceRelationshipsFromYaml(connection);
+    const std::set<core::Relationship>& relationships = connection->getRelationships();
+    REQUIRE(expectations.size() == relationships.size());
+    for (const auto& expected_relationship_name : expectations) {
+      const auto relationship_name_matches = [&] (const core::Relationship& relationship) { return expected_relationship_name == relationship.getName(); };
+      const std::size_t relationship_count = std::count_if(relationships.cbegin(), relationships.cend(), relationship_name_matches);
+      REQUIRE(1 == relationship_count);
+    }
+  }
+  SECTION("Queue size limits are read") {
+    YAML::Node connection_node = YAML::Load(std::string {
+        "max work queue size: 231\n"
+        "max work queue data size: 12 MB\n" });
+    YamlConnectionParser yaml_connection_parser(connection_node, "test_node", parent_ptr, logger);
+    REQUIRE(231 == yaml_connection_parser.getWorkQueueSizeFromYaml());
+    REQUIRE(12582912 == yaml_connection_parser.getWorkQueueDataSizeFromYaml());  // 12 * 1024 * 1024 B
```

Comment:
we could move the comment and let the compiler do the multiplication

Reply:
Maybe I am wrong, but I have the gut feeling that this compiles ever-so-slightly quicker. I will change it if you convince me there is no compile-time penalty :D

Reply:
leave it if you can tell me what `64 * 1024 * 1024` is faster than the compiler, it is about having a single source of truth, and in this case we could express in code, what is written in comment (me moving ~~in my chair~~ the cursor during compilation probably has a greater effect on the compilation time than this multiplication)

Reply:
I do not think you not-liking comments because they can lie is a good excuse for asking this change. Using comments for explaining a calculation is my choice of style (and I gave you the reason, I think it theoretically - albeit rather slowly - might increase compilation time).

Why do you think as the author of this PR, I should follow your favourite way of doing things, when the current proposal is just as expressive? Are you not being a bit arrogant?

Reply:
> But remember: while comments are very important, the best code is self-documenting.

from [this](https://google.github.io/styleguide/cppguide.html#Comments)

Reply:
Please remember the Apache motto "Community Over Code". (from https://www.apache.org/theapacheway/)
Good code is high priority but keeping a healthy community is higher. We should be open to constructive criticism, but also accept when we cannot reach a consensus and not sacrifice the community morale on the altar of tiny technical disagreements.

My decision making preference is: consensus > healthy debate > majority vote.

Reply:
This is how I solved the same problem some time ago: https://github.com/apache/nifi-minifi-cpp/pull/875/files#diff-42dd8bec1f7b514486d8de629c9a608eR64

Take what you like.

Reply:
But the code \*is\* self-documenting, the number it shows is obvious. I would definitely change this if I saw any chance that anyone who reads this couldn't immediately understand what the number means from the context.

Just for reference, I asked my girlfriend who has nothing to do with coding what it could mean that we read in 12 MB and get the number 12582912 and she could immediately tell that this is probably the number of bytes.

However, I decided to go further than this, and add an explanation, and my choice of style is a comment for the reason mentioned above. I call this 100% bikeshedding: https://www.wikiwand.com/en/Law_of_triviality